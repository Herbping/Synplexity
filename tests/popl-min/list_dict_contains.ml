type nat = 
  | Z
  | S of nat

type list =
  | Nil
  | Cons of (nat * nat) * list

type bool =
  | False
  | True

let rec list_dict_contains : (nat -> nat -> bool) -> nat -> list -> bool |>
  /\(0 -> 0 -> True,
     0 -> S nat -> False,
     S nat -> 0 -> False) ->
  0 ->
  /\(
        [] -> False,
        [(0, nat)] -> True,
        [(S nat, nat)] -> False,
        [(nat, nat); (0, nat)] -> True
    )
= ?
