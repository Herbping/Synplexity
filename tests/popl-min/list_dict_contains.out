(* Generated Wed Nov 11 22:01:33 2015 *)
fix f1 (f2:nat -> nat -> bool) : nat -> list -> bool ->
  fix f3 (n1:nat) : list -> bool ->
    fix f4 (l1:list) : bool ->
      match l1 with
      | Nil u1 -> False
      | Cons t1 -> match f4 (#2 t1) with
        | False u1 -> (f2 n1) (#1 (#1 t1))
        | True u1 -> True
